### ISSUE TITLE
linksection + @typeInfo = index out of bounds

### ISSUE THREAD

**Issue by garrettlennoxbeck** [ziglang/zig#18104]
### Zig Version

0.12.0-dev.1717+54f4abae2

### Steps to Reproduce and Observed Behavior

Replace the default (`zig init`) `src/main.zig` with the following and run `zig build`:
```
const foo linksection("foo_section") = 0;
// The section names can be the same, or different, both still crash.
const bar linksection("bar_section") = 0;

pub fn main() !void {
    _ = @typeInfo(@This());
}
```
The compiler crashes in `Zir.zig` with an `index out of bounds: index 1296779325, len 40` error.

I will note that the compiler only crashes if there are two or more `linksection` variables; remove either `foo` or `bar` from the previous example, and the issue goes away.
### Expected Behavior

The compiler should not crash.

**Comment by Vexu** [comment]
Probably fixed by #18054

**Comment by tw4452852** [comment]
Confirmed the above patch fixes this issue.
